pokemon/src/main/java/gamarket/Grid.java

```diff
@@ -144,13 +163,103 @@ private boolean canMove (int x, int y) {
         return grid;
     }
     /**
-     * change Tile at x, y to a new type
+     * Change Tile at x, y to a new type
      * @param x valid x cordinate
      * @param y valid y cordinate
-     * @param newType
+     * @param newType must be valid or will assign Unrecognized 
      */
 	public void changeTile(int x, int y, Type newType) {
-        Tile tileAtXY = this.gridState.get(x).get(y);
-        tileAtXY.setType(newType);
+        getTile(x,y).setType(newType);
+    }
+
+    /**
+     * Utility Function for retrieving Tiles at X,Y 
+     * @param x valid x grid coordinate 
+     * @param y valid y gird coordinate
+     * @return The tile at X,Y
+     */
+    public Tile getTile(int x, int y) {
+		return this.gridState.get(x).get(y);
 	}
+    
+    /**
+     * Converts grid to a string and saves to the given filename.
+     * Appends _gridData.txt to filename
+     * @param saveToFileName - name wanted for gridData file
+     */
+    public void save(String saveToFileName) {
+		File file = new File(saveToFileName + "_gridData.txt");
+		try {
+			FileWriter writer = new FileWriter(file);
+			
+			writer.write(this.toString());
+			
+			//writer.write(str);
+			writer.flush();
+			writer.close();
+		} catch (Exception e) {
+			System.out.println(e);
+		}
+    }
+
+    /**
+     * loads and transforms current grid to data for filename given to
+     * function. Appends _gridData.txt to filename to help gitignore
+     * assumes that file exists and is properly formatted.
+     * Proper Fomat 
+     * Grid demensions:xMax,yMax\n
+     * Player position is:playerX,playerY\n
+     * The grid State:\n
+     * tileValue grid space separated 
+     *  @param savedFileName - name of file wish to retreive
+     */
```

Comment:
Maybe include a savdFileName_gridData.txt example file to make it easier to follow along with the code in loadData